pangeo_forge_recipes/patterns.py

```diff
@@ -297,7 +297,15 @@ def pattern_blockchain(pattern: FilePattern) -> List[bytes]:
     # index:filepath pairs yielded by iterating over ``.items()``. if these pairs are generated in
     # a different way in the future, we ultimately don't care.
     root = {
-        "fsspec_open_kwargs": pattern.fsspec_open_kwargs,
+        "fsspec_open_kwargs": {
+            k: v
+            for k, v in pattern.fsspec_open_kwargs.items()
+            # `client_kwargs` is excluded because it can contain
+            # hard-to-hash objects, e.g. aiohttp.ClientTimeout.
+            # This avoids subsequent errors in
+            # `serialization.either_encode_or_hash()`.
+            if k != "client_kwargs"
```

Comment:
Could there be the `ssl` keyword included in this filtering? It causes similar troubles during hashing and my guess is that it's a similar issue: #418 ?

Reply:
It may be easiest to exclude `fsspec_open_kwargs` itself from the hash computation, rather than filtering a key list containing `client_kwargs` etc. @cisaacstern, what do you think?

I had a quick look at the stack trace in https://github.com/pangeo-forge/pangeo-forge-recipes/issues/418#issuecomment-1267066912 and it looks like a separate problem related to pickle. The expected error associated with this [PR issue](https://github.com/pangeo-forge/pangeo-forge-recipes/issues/443) should be raised in `pangeo-forge-recipes.serialization.either_encode_or_hash()`:

https://github.com/pangeo-forge/pangeo-forge-recipes/blob/fe5d1f7241e724fde76f86ce582fb0b830af3036/pangeo_forge_recipes/serialization.py#L25

e.g. 

```python
Input In [5], in either_encode_or_hash(obj)
     13 elif isinstance(obj, bytes):
     14     return obj.hex()
---> 15 raise TypeError(f"object of type {type(obj).__name__} not serializable")

TypeError: object of type ClientTimeout not serializable
```